enterprise/api/v1/views/enterprise_group.py

```diff
@@ -239,6 +244,14 @@ def assign_learners(self, request, group_uuid):
                 'new_learners': total_new_users_processed,
                 'existing_learners': total_existing_users_processed,
             }
+            if budget_expiration is not None and catalog_uuid is not None:
+                for membership in memberships:
```

Comment:
you could use `utils.batch()` here to break these memberships up into lists to send off to your async task.